src/core/services/theming/colors.js

```diff
@@ -0,0 +1,129 @@
+(function () {
+    "use strict";
+
+    //module global to keep an reference to the dynamic style sheet
+    var customSheet;
+    //module global to available color pallettes
+    var colorStore     = {};
+    var defaultPalette = '';
+
+/**
+ * @ngdoc service
+ * @name $mdColors
+ * @module material.core.color
+ *
+ * @description
+ * `$mdColors` makes it possible to use the colors of any palette to be available
+ *  as CSS rules in the form of md-bg-[colorname] and md-fg-[colorname] (as foreground and background)
+ *  If you want to activate another palette you can use the $mdColor.loadPalette(palletteName) method.
+ *
+ *  the $mdColors.palettes array lists all available palettes
+ *  the $mdColors.palette property lists the current selected palette
+ *  the $mdColors.colors array lists all the available colors with hex encoded color values for the current selected palette.
+ *
+ **/
+
+
+    angular
+        .module('material.core.colors',['material.core.theming'])
+        .service('$mdColors', mdColors)
+        .config(configColors)
+        .run(loadDefaults);
+
+    configColors.$inject = ['$mdThemingProvider'];
+    function configColors($mdThemingProvider) {
+        // fetch the colors out of the theming provider
+        Object.keys($mdThemingProvider._PALETTES).forEach(parsePalette);
+
+        // I need access to the default 'default', hardcoded for now as I miss access to it.
+        defaultPalette = $mdThemingProvider._THEMES['default'].colors.primary.name;
+        return;
+
+        // clone the palette colors to the colorStore var
+        function parsePalette(paletteName) {
+            var palette = $mdThemingProvider._PALETTES[paletteName];
+            var colors  = [];
+            colorStore[paletteName] = colors;
+            Object.keys(palette).forEach(copyColors);
+            return ;
+
+            function copyColors(colorName) {
+                // use an regex to look for hex colors, ignore the rest
+                if (/#[0-9A-Fa-f]{6}|0-9A-Fa-f]{8}\b/.exec(palette[colorName])) {
+                    colors.push({color:colorName,value:palette[colorName]});
+                }
+            }
+        }
+    }
+
+    loadDefaults.$inject=['$mdColors'];
+    function loadDefaults($mdColors) {
```

Comment:
Actually, I didn't mean load the palette in the provider. What I meant was not load a palette until the user instantiates the `$mdColors` service explicitly.
Thinking about it again, there is a design desicion to be made:

1. Automatically instantiate `$mdColors` and load the default palette.
2. Have the user explicitly load a palette if they want to use `$mdColors`.

With (1) there is less boilerplate for people that do want to use `$mdColors`, but there is unnecessary code run for people that don't care to use it (e.g. creating a stylesheet that is never ging to be used).
With (2) we avoid unnecessary initialization code, but require the user to "manually" load a palette if thy want to use it.

(Not sure which one I prefer. It would be good to somehow benchmark the initialization code).

--
BTW, with the current implementation, can there only be one palette active at a time ?
Wouldn't it be nice to be able to use more than one ?